apps/src/templates/progress/GoogleClassroomShareButton.jsx

```diff
@@ -27,17 +28,39 @@ class GoogleClassroomShareButton extends React.Component {
     itemtype: PropTypes.string.isRequired,
     title: PropTypes.string,
     height: PropTypes.number,
-    courseid: PropTypes.number
+    courseid: PropTypes.number,
+    analyticsData: PropTypes.object
   };
 
   constructor(props) {
     super(props);
     this.loadApi = this.loadApi.bind(this);
     this.waitForGapi = this.waitForGapi.bind(this);
     this.renderButton = this.renderButton.bind(this);
+    this.onShareStart = this.onShareStart.bind(this);
+    this.onShareComplete = this.onShareComplete.bind(this);
+    this.logEvent = this.logEvent.bind(this);
+
+    // For metrics in IE.
+    this.blur = this.blur.bind(this);
+    this.mouseOver = this.mouseOver.bind(this);
+    this.mouseOut = this.mouseOut.bind(this);
+    this.iframeMouseOver = false;
   }
 
   componentDidMount() {
+    // Use unique callback names since we're adding to the global namespace
+    window[this.onShareStartName()] = this.onShareStart;
+    window[this.onShareCompleteName()] = this.onShareComplete;
+
+    // The button callbacks are not supported in IE, so in IE we use click
+    // events to record analytics. The button is rendered in an iframe though,
+    // so to detect the click events we need to use the 'blur' event.
+    const isIE = /*@cc_on!@*/ false || !!document.documentMode;
```

Comment:
fyi: we've got a util function for this: https://github.com/code-dot-org/code-dot-org/blob/7e16615a0ee0dfc1105fd6a2a03ee3cf889dd427/apps/src/util/browser-detector.js#L86

Reply:
👍